synapse/storage/databases/main/stream.py

```diff
@@ -1200,8 +1200,6 @@ def _paginate_room_events_txn(
             `to_token`), or `limit` is zero.
         """
 
-        assert int(limit) >= 0
-
```

Comment:
Unneeded because it's enforced at construction time of a PaginationConfig?

I note that this function lets `limit` default to `-1`. Is that sensible/correct?

Reply:
A bunch of these functions have defaults for `limit` (and `dir`) because earlier arguments also have defaults, they don't really make sense though and I'm not really sure what to do about it.